src/index.js

```diff
@@ -146,12 +146,18 @@ export let ContextTree = _.curry(
         onError(error) // Raise the onError event
       }
     }
-    let triggerImmediateUpdate = F.debounceAsync(1, runUpdate)
-    let triggerDelayedUpdate = F.debounceAsync(debounce, runUpdate)
+
+    let triggerImmediatePathUpdate = _.memoize(() =>
```

Comment:
why do we need to memoize these? This only runs once when you instantiate a client instance anyway

Reply:
we need to isolate debouncing for different paths
you if you refresh root and then unpause a facet, second update will bounce out the root refresh
so using memo for different debouncers

Reply:
Let's just add a comment here explaining this